server/src/main/java/org/apache/iotdb/db/utils/FilePathUtils.java

```diff
@@ -50,36 +51,84 @@ private FilePathUtils() {
 
   /**
    * Format file path to end with File.separator
+   *
    * @param filePath origin file path
    * @return Regularized Path
    */
-  public static String regularizePath(String filePath){
+  public static String regularizePath(String filePath) {
     if (filePath.length() > 0
         && filePath.charAt(filePath.length() - 1) != File.separatorChar) {
       filePath = filePath + File.separatorChar;
     }
     return filePath;
   }
 
+  /**
+   * IMPORTANT, when the path of TsFile changes, the following methods should be changed
+   * accordingly. The sequence TsFile is located at ${IOTDB_DATA_DIR}/data/sequence/. The unsequence
+   * TsFile is located at ${IOTDB_DATA_DIR}/data/unsequence/. Where different storage group's TsFile
+   * is located at <logicalStorageGroupName>/<virtualStorageGroupName>/<timePartitionId>/<fileName>.
+   * For example, one sequence TsFile may locate at /data/data/sequence/root.group_9/0/0/1611199237113-4-0.tsfile
+   *
+   * @param resource the tsFileResource
```

Comment:
UT is important for this class, as it is very important to guarantee it works on all platforms.

In some degree, IT is also useful. If we have IT, then we can guarantee other committers have to modify this function if they modified the position of the tsfiles.
E.g., start IoTDB -> write some data -> flush -> split the real data file path.

Reply:
By the way, is it necessary to get `getAbsolutePath`? (Seems you only use the last 3 level of folder names) (I am not sure)

Reply:
Ok, I'll and some UT and IT.

> By the way, is it necessary to get getAbsolutePath? (Seems you only use the last 3 level of folder names) (I am not sure)
Yes, we need to get the last 4 level of the folder name, so `getAbsolutePath` maybe is the most appropriate method